src/rust/engine/src/nodes.rs

```diff
@@ -1205,13 +1205,9 @@ impl Node for NodeKey {
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
     let span_id = generate_random_64bit_string();
-    let node_workunit_params = if context.session.should_record_zipkin_spans() {
```

Comment:
Can you explain why we're not checking anymore that "`session.should_record_zipkin_spans`"? (I don't see a self-explanatory reason in the rest of the code but may have missed something)

Reply:
We just always record, regardless of whether should_record_zipkin_spans is set.

Reply:
Yes, but why is that desired?

Reply:
I've added back the check for this